#! /usr/bin/env python

PACKAGE='tuw_self_localization'
from dynamic_reconfigure.parameter_generator_catkin import *
gen = ParameterGenerator()

gen.add("forward_prediction_time", double_t, 0, "forward_prediction_time", 0.1, 0.1, 10)
gen.add("init_sigma_location", double_t, 0, "init_sigma_location", 0.5, 0, 20)
gen.add("init_sigma_orientation", double_t, 0, "init_sigma_orientation", 0.39, 0, 3.14)

gen.add("enable_prediction", bool_t,   0, "enables prediction step",  True)
gen.add("enable_data_association", bool_t,   0, "enables data_association step",  True)
gen.add("enable_correction", bool_t,   0, "enables correction step",  True)

gen.add("alpha_1", double_t, 0, "motion noise alpha 1", 3, 0, 10)
gen.add("alpha_2", double_t, 0, "motion noise alpha 2", 0.5, 0, 6.28)
gen.add("alpha_3", double_t, 0, "motion noise alpha 3", 2, 0, 10)
gen.add("alpha_4", double_t, 0, "motion noise alpha 4", 0.2, 0, 6.28)
gen.add("sigma_rho", double_t, 0, "measurment on lines rho", 1, 0, 10)
gen.add("sigma_alpha", double_t, 0, "measurment on lines alpha", 0.1, 0, 3.14)


gen.add("plot_hough_space",    bool_t,   0, "",  True)
gen.add("hough_space_pixel_rho", double_t, 0, "hough_space_size_rho", 600, 0, 1000)
gen.add("hough_space_pixel_alpha", double_t, 0, "hough_space_size_alpha", 600, 0, 1000)
gen.add("hough_space_meter_rho", double_t, 0, "hough_space_resolution_rho", 5, 0, 20)

gen.add("data_association_line_rho", double_t, 0, "threshold for line matching", 0.5, 0, 1)
gen.add("data_association_line_alpha", double_t, 0, "threshold for line matching", 0.4, 0, 1.57)
gen.add("data_association_distance_to_endpoints", double_t, 0, "squared distance to endpoints", 4, 0, 5)

gen.add("line_dection_split_neighbor", bool_t, 0, " ", True)
gen.add("line_dection_split_threshold", double_t, 0, "Split threshold", .05, 0, 0.5)
gen.add("line_dection_min_length", double_t, 0, "Min line length", 1, 0, 5)
gen.add("line_dection_min_points_per_line", int_t, 0, "Minimum points on line", 40, 0, 100)
gen.add("line_dection_min_points_per_unit", double_t, 0, "Minimum number of points per unit", 10, 0, 100)

exit(gen.generate(PACKAGE, "tuw_self_localization", "KalmanFilter"))
